xmind/tests/test_workbook_element.py

```diff
@@ -0,0 +1,751 @@
+        _getOwnerWorkbook.assert_called_once_with()
+        self.assertListEqual([
+            call(11, None),
+            call(12, None),
+            call(13, None)], _sheet_element_constructor.call_args_list)
+        self.assertListEqual(_sheets, ['a', 'b', 'c'])
+
+        _patcher.stop()
+
+    def test_WorkbookElement_getSheet_is_sheets_is_ownerbook(self):
+        _test_el = WorkbookElement()
+        self._remove_patched_function('_getChildNodesByTagName')
+        self._remove_patched_function('_getOwnerWorkbook')
+        self._remove_patched_function('_sheet_element_init')
+
+        _getChildNodesByTagName = self._init_patch_with_name(
+            '_getChildNodesByTagName', 'xmind.core.Node.getChildNodesByTagName', return_value=[11, 12, 13])
+        _getOwnerWorkbook = self._init_patch_with_name(
+            '_getOwnerWorkbook', 'xmind.core.mixin.WorkbookMixinElement.getOwnerWorkbook',
+            return_value='workbook')
+
+        _patcher = patch('xmind.core.workbook.SheetElement')
+        _sheet_element_constructor = _patcher.start()
+        _sheet_element_constructor.side_effect = ['a', 'b', 'c']
+
+        _sheets = _test_el.getSheets()
+
+        _getChildNodesByTagName.assert_called_once_with(TAG_SHEET)
+        _getOwnerWorkbook.assert_called_once_with()
+        self.assertListEqual([
+            call(11, 'workbook'),
+            call(12, 'workbook'),
+            call(13, 'workbook')], _sheet_element_constructor.call_args_list)
+        self.assertListEqual(_sheets, ['a', 'b', 'c'])
+
+        _patcher.stop()
+
+    def test_WorkbookElement_getSheetByIndex_index_none(self):
+        _test_el = WorkbookElement()
+
+        _getSheets = self._init_patch_with_name(
+            '_getSheets', 'xmind.core.workbook.WorkbookElement.getSheets',
+            return_value=['sheet1', 'sheet2', 'sheet3'])
+
+        _sheets = _test_el.getSheetByIndex(None)
+
+        _getSheets.assert_called_once_with()
+        self.assertEqual(_sheets, None)
+
+    def test_WorkbookElement_getSheetByIndex_index_less_0(self):
+        _test_el = WorkbookElement()
+
+        _getSheets = self._init_patch_with_name(
+            '_getSheets', 'xmind.core.workbook.WorkbookElement.getSheets',
+            return_value=['sheet1', 'sheet2', 'sheet3'])
+
+        _sheets = _test_el.getSheetByIndex(-1)
+
+        _getSheets.assert_called_once_with()
+        self.assertEqual(_sheets, None)
+
+    def test_WorkbookElement_getSheetByIndex_index_equal_len_sheets(self):
+        _test_el = WorkbookElement()
+
+        _getSheets = self._init_patch_with_name(
+            '_getSheets', 'xmind.core.workbook.WorkbookElement.getSheets',
+            return_value=['sheet1', 'sheet2', 'sheet3'])
+
+        _sheets = _test_el.getSheetByIndex(3)
+
+        _getSheets.assert_called_once_with()
+        self.assertEqual(_sheets, None)
+
+    def test_WorkbookElement_getSheetByIndex_index_greater_then_len_sheets(self):
```

Comment:
I would consolidate 4 tests in one 
_test_el.getSheetByIndex(None)
_test_el.getSheetByIndex(-1)
_test_el.getSheetByIndex(3)
_test_el.getSheetByIndex(4) 

and just check in the end that _getSheets has been called 4 times